examples/green-house-migration/routes/api/prospects.py

```
@@ -0,0 +1,2573 @@
+        if "data" not in response:
+            raise HTTPException(
+                status_code=400, detail="Failed to update prospect pool"
+            )
+
+        return _normalize_prospect_pool(response["data"])
+
+    except Exception as e:
+        logger.error("Failed to update prospect pool %s: %s", pool_id, e)
+        raise HTTPException(
+            status_code=500, detail="Failed to update prospect pool: {str(e)}"
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)